tools/crm_diff.c

```diff
@@ -207,8 +186,13 @@ generate_patch(xmlNode *source, xmlNode *target, bool as_cib, bool no_version)
     }
 
     pcmk__log_xml_patchset(LOG_NOTICE, patchset);
-    print_xml(patchset);
+
+    buffer = g_string_sized_new(1024);
+    pcmk__xml_string(patchset, pcmk__xml_fmt_pretty, buffer, 0);
+    out->output_xml(out, PCMK_XE_PATCHSET, buffer->str);
```

Comment:
I'm fine with revisiting what `output_xml` takes as an argument.  I think the reason it takes a string at the moment is likely due to whatever called it first having an XML string already.  Over time, we've added more and more callers and it looks like almost all of them now have an xmlNode instead.  I'm not willing to dig into the history to figure that out, so it's just a guess.